src/java/org/apache/cassandra/config/YamlConfigurationLoader.java

```diff
@@ -199,6 +239,7 @@ public static <T> T fromMap(Map<String,Object> map, boolean shouldCheck, Class<T
     {
         Constructor constructor = new YamlConfigurationLoader.CustomConstructor(klass, klass.getClassLoader());
         Map<Class<?>, Map<String, Replacement>> replacements = getNameReplacements(Config.class);
+        verifyReplacements(replacements, map);
```

Comment:
when you rebase you will find a new method

```
public static <T> T updateFromMap(Map<String, ? extends Object> map, boolean shouldCheck, T obj)
```

Can you add this same logic there as well?  replacement is there + same style map, so just need this one line; cool doing this on merge